components/modal/store/purchase/GiftModal.tsx

```diff
@@ -1,28 +1,45 @@
-import { useEffect, useState } from 'react';
-import { Gift } from 'types/itemTypes';
+import { useState } from 'react';
+import { useSetRecoilState, useResetRecoilState } from 'recoil';
+import { GiftRequest } from 'types/itemTypes';
 import { PriceTag } from 'types/modalTypes';
+import { instance } from 'utils/axios';
+import { errorState } from 'utils/recoil/error';
+import { modalState } from 'utils/recoil/modal';
 import {
   ModalButtonContainer,
   ModalButton,
 } from 'components/modal/ModalButton';
 import GiftSearchBar from 'components/store/purchase/GiftSearchBar';
-import useGiftModal from 'hooks/modal/store/purchase/useGiftModal';
 import styles from 'styles/modal/store/GiftModal.module.scss';
 
 export default function GiftModal({ itemId, product, price }: PriceTag) {
-  const [recipient, setRecipient] = useState<string>('');
-  const [gift, setGift] = useState<Gift>({
-    itemId: -1,
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const resetModal = useResetRecoilState(modalState);
+  const setError = useSetRecoilState<string>(errorState);
+  const [giftReqData, setGiftReqData] = useState<GiftRequest>({
     ownerId: '',
   });
-  const { onPurchase, onCancel } = useGiftModal(gift);
 
-  useEffect(() => {
-    setGift({
-      itemId: itemId,
-      ownerId: recipient,
-    });
-  }, [itemId, recipient]);
+  // TODO: 에러 처리
+  const onPurchase = async () => {
+    if (giftReqData.ownerId === '') {
+      alert('선물할 유저를 선택해주세요.');
+      return;
+    }
+    setIsLoading(true);
+    try {
+      const res = await instance.post(
+        `/pingpong/items/gift/${itemId}`,
+        giftReqData
+      );
+      if (res.status === 201) {
+        alert(`${giftReqData.ownerId}님께 선물이 전달되었습니다.`);
+      }
```

Comment:
여기 `res.status`가 201인 경우에는 처리가 되어서 `alert`가 뜰 것 같은데, 200같은 다른 성공 응답이 혹시라도 온다면 이 부분에 대해서는 처리가 안되어 있어서 바로 모달이 꺼질 것 같아요...!
특별히 `res.status` 201에 대한 처리가 필요한 게 아니라면 그냥 `catch` 블록으로 넘어가지 않았을 경우에는 성공이라고 판단하고 `alert`를 띄우는 것도 괜찮지 않을까 싶습니다 ^_ㅜ

```suggestion
      const res = await instance.post(
        `/pingpong/items/gift/${itemId}`,
        giftReqData
      );
      alert(`${giftReqData.ownerId}님께 선물이 전달되었습니다.`);
```

Reply:
그렇게 하는게 좋을 것 같네요! 수정하겠습니다👍